### PR TITLE
 Week 3 Assessment complete

### PR DESCRIPTION
**Pull request by TKash89**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Interview Questions
- [x] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
I would appreciate some feedback on Challenge Q # 3. I must have been jumbling up the .map() HOF. I felt like I had the right idea somehow.  Thank you for your time.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -12,20 +12,67 @@
...
+                  //           Tests:       1 failed, 1 total
+
 
 // b) Create the function that makes the test pass.
 
+    
+let fibArr = (array) => {
+  for(let i= 2; i< fibLength1.length; i++);
+  for(let i= 2; i< fibLength2.length; i++);
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:49]: So I guess my first question here is why you didn't make a function named sequenceNums if that was your test function name? 

To bring this up another level you will want to record the output of this function and what you would need to do next to solve for it.

File: code-challenges.test.js [hunk 2]
```diff
@@ -51,10 +98,36 @@ const studyMinutesWeek2 = {
 
 // b) Create the function that makes the test pass.
 
+
+const studious = () => {
+  return  Object.values(studyMinutesWeek1.sort())
+  return  Object.values(studyMinutesWeek2.sort())
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:105]: Remember a function can only have one return.

File: code-challenges.test.js [hunk 3]
```diff
@@ -64,4 +137,11 @@ const accountTransactions2 = [250, -89, 100, -96]
...
 
-// b) Create the function that makes the test pass.
+// b) Create the function that makes the test pass
+
+const addAccounts = (array) => {
+  return array.map(accountTransactions1[0+1+2+3])
+  return array.map(accountTransactions2[0+1+2+3])
+  return array.map(accountTransactions3[0+1+2+3])
+
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:147]: So yeah if you check out the [syllabus section on map](https://github.com/learn-academy-2022-golf/Syllabus/blob/main/javascript/higher-order-functions.md#map) you'll find that it requires usage of an anonymous function as an argument, called a callback function. This function has it's own parameters and logic that need to be passed in to allow for a return per item of the array.

File: interview-questions.md [hunk 4]
```diff
@@ -8,48 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Your answer:
+Your answer: 
+                JSX, in my opinion, is like a "hybrid language" that is used in REACT. React is the facebook/meta created javascript library used to build singular page applications on the web. A significant syntax difference between these two is that JSX allows javascript to be written along side HTML in a file. Without this you would be required to use seperate files, then link them together.
 
 Researched answer:
+                JSX is the language used in React meaning Javascript XML, or Javascript Syntax Extension. It combines the features of both Javascript and HTML languages, using components instead of completely seperate files in order to efficiently use react software. The difference between HTML and JSX is that when writting HTML, you are not required to return a single parent element. Using JSX, returning a parent element is needed or else it will not compile. On another note, one last difference is that you cannot write any JS code in HTML, unless you properly place a script tag linking them together. In JSX you are free to travel to Javascript land anytime of your choosing.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:15]: This is a great answer!

File: interview-questions.md [hunk 5]
```diff
@@ -8,48 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 2. What is yarn? What file(s) are modified in a React application when you run the command yarn in your terminal?
 
 Your answer:
+                To the best of my knowledge, yarn is software containing many nodes and snippets that provide tools for React, and to an extent, Jest. I'm honestly not comfortable with the exact nomenclature, however, I'd like to take the liberty and research more about yarn, to provide a much better answer.
 
 Researched answer:
+                Yarn is large and relitively new package manager that was created to facilitate avoiding the rewriting of common pieces of code. In collaboration between Tilde, Google, and Exponent, it is an open source package manager in which engineers have shared thousands of snippets of code. This has enabled developers access to over 300,000 packages.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:23]: Solid!

File: interview-questions.md [hunk 6]
```diff
@@ -8,48 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 5. What is the DOM?
 
-Your answer:
+Your answer: 
+                Unfortunetly, this question has got the best of me! I'm having a difficult time stating the exact definition. That being said, I am more than happy to give a conceptual explaining the best I can: The DOM, which is an acronym, is the visual display that appears from the written React JSX code. This is where the user interface would be.
 
 Researched answer:
+                I was off on my summarization regarding what DOM means. The DOM stands for document object model, and you can think of it as a tree structure that represents the web page itself. This is incredibly useful when building a web page in React because it allows the designer to change content in Javascript easier than sifting through code. Anytime there are changes in React, it updates the DOM as well.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:47]: Awesome correction!

File: react-debugging.html [hunk 7]
```diff
@@ -63,20 +69,17 @@
...
+          <button> onClick= Click to Draw a Cardbutton></button>
+          <button> onClick={shuffle}>
+            Return Cards and Shuffle Deck></button>
+          <div> style={handStyles}>
             {hand.map((card, index) => {
               return (
-                <div style={cardStyles} key={index}>
-                  {card}
+                <div> style={cardStyles} key={index}
+                  {card}>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:82]: Looks like there was a bit of struggle with this section. What process did you follow when evaluating the errors?

**Review by WalkerTexas-coder — APPROVED** [review]
Tyler I want to say thanks for putting in the effort on this assessment. Your interview answers are great and I think that they showcase a knowledge set that is growing rapidly. Personally this weeks assessment was the hardest for me. 

This assessment is currently at the novice level. I want to encourage you to head back to the syllabus when you are working on your code challenges and make sure you are using tools correctly. If you are making a function look up the function section, a test the test section, a map utilize the HOF's section. If you spend the time now it will pay off in the long run. 

I have asked a few questions below and responded to your feedback as well. Feel free to answer me or not but I would like to know your thoughts.

**Comment by TKash89** [comment]
Austin,

            As far as the for loop, the sequenceNums was not placed in there because I totally looked past it after I made it. I know it probably would've put me closer towards a solution. That was a tough problem, and I understand that there is usually one line or two in order to solve these difficult ones.
            The debugging solution I found myself heavily focusing on syntax around the buttons and divs. Only to find out earlier the big fix was placing the <> </> around the problem to unbreak it.

Tyler
________________________________

**Pull request merged**